airflow_spark_on_k8s_job_builder/core.py

```diff
@@ -271,18 +271,18 @@ def set_driver_cores(self, cores: int) -> "SparkK8sJobBuilder":
         if not cores:
             raise ValueError("Need to provide a non-empty value for the number of driver cores")
         self.get_job_params()["driver"]["cores"] = cores
-        min_max_cores = self._cast_cores_to_int(self.get_driver_cores(), self.get_driver_cores_limit(), "driver")
-        if min_max_cores[0] is not None and min_max_cores[1] is not None and min_max_cores[0] > min_max_cores[1]:
+        [requested_cores, max_cores] = self._cast_cores_to_int(self.get_driver_cores(), self.get_driver_cores_limit(), "driver")
+        if requested_cores is not None and max_cores is not None and requested_cores > max_cores:
             self.set_driver_cores_limit(cores)
         return self
 
     def get_driver_cores_limit(self):
-        return self.get_job_params()["driver"]["coreLimit"]
+        return self.get_job_params()["driver"].get("coreLimit", None)
```

Comment:
minor nit: think there's no point in setting `None` as a default ;)

Reply:
Nice! for some reason I had it in my head that without a default it goes to `KeyError`, but you are right, it returns `None`